packages/contextual-help/src/ContextualHelp.ts

```diff
@@ -0,0 +1,205 @@
+
+    @state()
+    overlayOpen = false;
+
+    private get buttonAriaLabel(): string {
+        if (this.label) {
+            return this.label;
+        } else {
+            if (this.variant === 'help') {
+                return 'Help';
+            }
+            return 'Informations';
+        }
+    }
+
+    private renderPopover(event: SlottableRequestEvent): void {
+        event.stopPropagation();
+
+        if (event.data === removeSlottableRequest) {
+            render(undefined, event.target as HTMLElement);
+            return;
+        }
+
+        import('@spectrum-web-components/popover/sp-popover.js');
+
+        const template = html`
+            <sp-popover class="popover">
+                <section>
+                    ${this.headline &&
+                    html`
+                        <h2 class="heading">${this.headline}</h2>
+                    `}
+                    <slot></slot>
+                    <div class="link">
+                        <slot name="link"></slot>
+                    </div>
+                </section>
+            </sp-popover>
+        `;
+
+        render(template, event.target as HTMLElement);
+    }
+
+    private renderDialog(event: SlottableRequestEvent): void {
+        event.stopPropagation();
+
+        if (event.data === removeSlottableRequest) {
+            render(undefined, event.target as HTMLElement);
+            return;
+        }
+
+        import('@spectrum-web-components/dialog/sp-dialog-wrapper.js');
+        const headlineVisibility = !this.headline ? 'none' : undefined;
+
+        const template = html`
+            <sp-dialog-wrapper
+                dismissable
+                underlay
+                headline=${ifDefined(this.headline)}
+                headline-visibility=${ifDefined(headlineVisibility)}
+            >
+                <slot></slot>
+                <div class="link">
+                    <slot name="link"></slot>
+                </div>
+            </sp-dialog-wrapper>
+        `;
+
+        render(template, event.target as HTMLElement);
+    }
+
+    protected override render(): TemplateResult {
+        const actualPlacement = this.isMobile.matches
+            ? undefined
+            : this.placement;
+
+        return html`
+            <sp-action-button
+                quiet
+                size="s"
+                id="trigger"
+                aria-label=${this.buttonAriaLabel}
+                ?is-active=${this.overlayOpen}
+            >
+                ${this.variant === 'help'
+                    ? html`
+                          <sp-icon-help-outline
+                              slot="icon"
+                          ></sp-icon-help-outline>
+                      `
+                    : html`
+                          <sp-icon-info-outline
+                              slot="icon"
+                          ></sp-icon-info-outline>
+                      `}
+            </sp-action-button>
+            <sp-overlay
```

Comment:
We may also want to look at the `trigger` directive here, as it lists as _slightly_ more performant that leveraging the element.

Reply:
I was not aware of this directive. I think it is really cool that is handles the `slottable-request` event for you.
A thing I did not found straight-forward is the open/close events. Is there a better way to know when it opens/closes events besides adding event listeners on the `ContextualHelp` class?
I'd need to know of open/close states for the button's `active` state. Would you consider a good addition to add `active` attribute on the trigger as a built-in feature of the `trigger` directive?

Reply:
As a consumer, the first thing I looked for was a `handleOpen` and `handleClose` callback, as `slottable-request` event is part of the internal works of this directive and I don't know if as a consumer I should be aware of that.

`sp-open`/`sp-close` events are fine, but are actually a bit delayed, because of the CSS transitions that have dispatched I guess. So the question with `handleOpen` / `handleClose` would be if the timing of calling these callbacks should be on the `handleSlottableRequest` method, or a bit late, on `sp-open` and `sp-closed`.

In my case as you mentioned `sp-open`/`sp-close` are a bit late :(

Reply:
Yes, `sp-open` and `sp-close` are not "correct for this situation.

If we went with callbacks, they would _likely_ happen at `slottable-request` time, because that represents the _outside_ of the open/close transition, rather than only the end of each transition. But, it could be argued that this belongs at the _beginning_ of each transition, which is when the state actually changes. 🤔 

Relying on the state feels nicer than relying on a knowable part of the lifecycle, which is why I like the `TriggerController`, but there is much to think about here.

Reply:
> that best I can think of with the established API is to use specific handing of slottable-request to toggle open which is rendered to active.

Regarding this, I don't think this works, as the event is attached on the `sp-overlay`, which I don't have access to.

Reply:
Indeed. So we can ship once and edit later leveraging `<sp-overlay>` directly, or we can hold this for an expansion of the directives capabilities. Do you have a preference on that?

Reply:
Depending on when exactly would the directive be expanded. If it's more than 1-2 releases, then I'd ship it as it is and come back when this behaviour is available.